scripts/deployment/upload_model_to_huggingface.py

```
@@ -0,0 +1,465 @@
+    print("  3. Upload it to your HuggingFace account")
+    print("  4. Update deployment configurations")
+    print()
+
+def find_best_trained_model() -> Optional[str]:
+    """Find the best trained model from common locations."""
+    print("🔍 SEARCHING FOR TRAINED MODELS")
+    print("=" * 40)
+    
+    # Ensure primary model directory exists
+    primary_model_dir = "/Users/minervae/Projects/SAMO--GENERAL/SAMO--DL/deployment/models"
+    if not os.path.exists(primary_model_dir):
```

Ruff (0.12.2)

`os.path.exists()` should be replaced by `Path.exists()`

(PTH110)